### PR TITLE
chore(repo): QoL for subpaths script

### PR DESCRIPTION
**Pull request by LekoArts**
## Description

Follow-up to https://github.com/clerk/javascript/pull/4172 - this makes `package.json#files` the source of truth for the subpaths.mjs scripts.

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

File: packages/localizations/subpaths.mjs [hunk 1]
```diff
@@ -1,45 +1,13 @@
...
-  'hr-HR',
-];
+/**
+ * This file is a helper for the "subpath-workaround.mjs" script.
+ * Add your new subpath to package.json#files.
+ *
+ * When you add an entry to the package.json "files" field, a subfolder will be automatically created with a package.json pointing to that file
+ */
+
+export const subpathNames = packageJson.files.filter(k => k !== 'dist');
```
Comment (LekoArts) [inline packages/localizations/subpaths.mjs:10]: Source of truth now:

https://github.com/clerk/javascript/blob/26ab88cd554d09531cd53859e18f7111689574a3/packages/localizations/package.json#L53-L92

File: packages/react/subpaths.mjs [hunk 2]
```diff
@@ -1,10 +1,13 @@
...
-  .filter(k => k.startsWith('./') && k !== './' && k !== './package.json')
-  .map(k => k.replace('./', ''));
+/**
+ * This file is a helper for the "subpath-workaround.mjs" script.
+ * Add your new subpath to package.json#files.
+ *
+ * When you add an entry to the package.json "files" field, a subfolder will be automatically created with a package.json pointing to that file
+ */
+
+export const subpathNames = packageJson.files.filter(k => k !== 'dist');
```
Comment (LekoArts) [inline packages/react/subpaths.mjs:10]: https://github.com/clerk/javascript/blob/26ab88cd554d09531cd53859e18f7111689574a3/packages/react/package.json#L59-L63

**Review by tmilewski — APPROVED** [review]
🔥

File: packages/shared/subpaths.mjs [hunk 3]
```diff
@@ -1,38 +1,13 @@
...
-  'pathToRegexp',
-];
+/**
+ * This file is a helper for the "subpath-workaround.mjs" script.
+ * Add your new subpath to package.json#files.
+ *
+ * When you add an entry to the package.json "files" field, a subfolder will be automatically created with a package.json pointing to that file
+ */
+
+export const subpathNames = packageJson.files.filter(k => !['dist', 'scripts', 'react'].includes(k));
```
Comment (LekoArts) [inline packages/shared/subpaths.mjs:10]: `scripts` should be left alone, `react` is a barrel file so handled below

**Pull request merged**